.github/workflows/docs.yml

```diff
@@ -11,15 +11,27 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: 'recursive'
-      - uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
-          python-version: "3.7"
-      - name: Install dependencies
+          python-version: "3.10"
+      - name: Bootstrap poetry
         run: |
-          python -m pip install --upgrade pip
-          pip install tox virtualenv
+          curl -sL https://install.python-poetry.org | python - -y ${{ matrix.bootstrap-args }}
+      - name: Configure poetry
+        run: poetry config virtualenvs.in-project true
+      - name: Set up cache
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Ensure cache is healthy
+        if: steps.cache.outputs.cache-hit == 'true'
+        run: timeout 10s poetry run pip --version || rm -rf .venv
```

Comment:
This can be simplified once `actions/setup-python` supports it natively: https://github.com/actions/setup-python/pull/281

Reply:
do you think this will be included in the setup-python action v3 or this is a minor change?

Reply:
They've added caching as non-breaking change, and this feels smaller and non-breaking as well, so to could be in v2, unless there is an unrelated breaking change before it's merged, of course.